server/services/store/sqlstore/sqlstore.go

```diff
@@ -20,12 +21,12 @@ type SQLStore struct {
 	dbType           string
 	tablePrefix      string
 	connectionString string
-	isPlugin         bool
+	pluginAPI        plugin.API
 	logger           *mlog.Logger
 }
 
 // New creates a new SQL implementation of the store.
-func New(dbType, connectionString, tablePrefix string, logger *mlog.Logger, db *sql.DB, isPlugin bool) (*SQLStore, error) {
+func New(dbType, connectionString, tablePrefix string, logger *mlog.Logger, db *sql.DB, pluginAPI plugin.API) (*SQLStore, error) {
```

Comment:
I don't like this, we are passing way too much information here. Can we receive here a mutexFactory function or can we receive something that matches an interface? I don't want to pass the whole API to the store.

Reply:
Addressed in https://github.com/mattermost/focalboard/pull/1776/commits/25377a965ebcd36d76acafd71ab62ab2c038989e, PTAL